### ISSUE TITLE
micronaut-hibernate-gorm - Upgrading to 3.0.0 causes NPE during startup with 'version-less' domains

### ISSUE THREAD

**Issue by mitushy** [micronaut-projects/micronaut-groovy#247]
### Expected Behavior
After upgrading to micronaut 3.0.0 (+ micronaut-gorm 3.0.0) applications fails to start if they contain 'version-less' domain objects. e.g:

```groovy
@Entity
class AccountRole implements GormEntity<AccountRole>, Serializable {
    Account account
    Role role

    Date dateCreated
    Date lastUpdated

    static constraints = {
        account nullable: false
        role nullable: false, lazy: false
    }

    static mapping = {
        id composite: ['account', 'role']
        table: 'account_role'
        version false                                 // <-- what we care about
    }
}
```

### Actual Behaviour

The application fails during startup. This looks to be caused by https://github.com/grails/gorm-hibernate5/issues/265

Example error:
```
Bean definition [org.grails.orm.hibernate.HibernateDatastore] could not be loaded: Error instantiating bean of type [org.grails.orm.hibernate.HibernateDatastore]: Could not instantiate persister org.hibernate.persister.entity.SingleTableEntityPersister
...
Caused by: java.lang.NullPointerException
	at org.hibernate.tuple.entity.EntityMetamodel.<init>(EntityMetamodel.java:398)
```

### Steps To Reproduce

1. Create a micronaut 3 app with groovy + gorm
2. Create a version-less entity

### Workaround

This can be worked-around by constraining the version in build.gradle:
```groovy
// Need 7.1.0.RC2 for now as previous versions fail when initializing non-version domains (e.g AccountRole)
// See https://github.com/grails/gorm-hibernate5/issues/265
constraints {
    implementation("org.grails:grails-datastore-gorm:7.1.0.RC2")
    implementation("org.grails:grails-datastore-gorm-hibernate5:7.1.0.RC2")
}
```

### Version

3.0.0

**Comment by mitushy** [comment]
Not sure if this merits an issue - but it took me a while to find the root cause and workaround, so opening one. This should be  solved for everyone once the [gorm deps](https://github.com/micronaut-projects/micronaut-groovy/blob/master/gradle.properties#L9) include the fix. RC1 and RC2 worked for me.

**Comment by mitushy** [comment]
Closing this. It has been resolved with the https://github.com/micronaut-projects/micronaut-groovy/releases/tag/v3.1.0 release, which will hopefully get included in the Micronaut 3.4.0+ BOM.
